src/providers/seventv/SeventvEmotes.cpp

```diff
@@ -362,4 +384,105 @@ void SeventvEmotes::loadChannelEmotes(const std::weak_ptr<Channel> &channel,
         .execute();
 }
 
+boost::optional<EmotePtr> SeventvEmotes::addEmote(
+    Atomic<std::shared_ptr<const EmoteMap>> &map,
+    const SeventvEventApiEmoteAddDispatch &dispatch)
+{
+    // Check for visibility first, so we don't copy the map.
+    auto emoteData = dispatch.emoteJson["data"].toObject();
+    if (emoteData.empty() || !checkEmoteVisibility(emoteData))
+    {
+        return boost::none;
+    }
+
+    EmoteMap updatedMap = *map.get();
+    auto result = createEmote(dispatch.emoteJson, emoteData, false);
+    if (!result.hasImages)
+    {
+        // this shouldn't happen but if it does it will crash
```

Comment:
Clarify comment, maybe something along the lines of:
`incoming Emote didn't contain any images, abort`